src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJCourse.java

```diff
@@ -16,27 +21,45 @@
   @NotNull
   private final APlusProject project;
 
+  public final CommonLibraryProvider commonLibraryProvider;
+
   public IntelliJCourse(@NotNull String name,
                         @NotNull List<Module> modules,
                         @NotNull List<Library> libraries,
                         @NotNull Map<String, String> requiredPlugins,
                         @NotNull Map<String, URL> resourceUrls,
                         @NotNull APlusProject project) {
-    super(name, modules, libraries, requiredPlugins, resourceUrls,
-        new CommonLibraryProvider(project));
+    super(name, modules, libraries, requiredPlugins, resourceUrls);
 
     this.project = project;
+    commonLibraryProvider = new CommonLibraryProvider(project);
   }
 
   @NotNull
   public APlusProject getProject() {
     return project;
   }
 
+  @Nullable
+  @Override
+  public Component getComponentIfExists(@NotNull String name) {
+    Component component = super.getComponentIfExists(name);
+    return component != null ? component : commonLibraryProvider.getComponentIfExists(name);
+  }
+
   @Nullable
   public Component getComponentIfExists(VirtualFile file) {
     Component component = getComponentIfExists(file.getName());
-    return component != null && component.getPath().toString().equals(file.getPath())
+    return component != null && component.getFullPath().equals(Paths.get(file.getPath()))
```

Comment:
do you think there could be ways to improve the readability of `component.getFullPath().equals(Paths.get(file.getPath()))`?

Reply:
`Paths.get(file.getPath()).equals(component.getFullPath())` ?

Reply:
I had some extraction and renaming in mind